packages/@lwc/synthetic-shadow/src/faux-shadow/node.ts

```diff
@@ -300,7 +300,16 @@ defineProperties(Node.prototype, {
             if (isNodeShadowed(this)) {
                 return parentNodeGetterPatched.call(this);
             }
-            return parentNodeGetter.call(this);
+
+            const parentNode = parentNodeGetter.call(this);
+
+            // Handle the case where a top level light DOM element is slotted into a synthetic
+            // shadow slot.
+            if (!isNull(parentNode) && isSyntheticSlotElement(parentNode)) {
```

Comment:
When will this be the case? `isSyntheticSlotElement` implementation is as:

```js
export function isSyntheticSlotElement(node: Node): node is HTMLSlotElement {
    return isSlotElement(node) && isNodeShadowed(node);
}
```

but if this line is executed, we know for sure that `isNodeShadowed(node)` is false, therefore `isSyntheticSlotElement` will always return `false`.